### PR TITLE
Change shape for filtered lollipops and subtracks

### PR DESCRIPTION
**Pull request by creilly8**
## Description
When a filterObj is present, an option to change the shape of the skewers appears in the variant menu. The user may select between eight different shapes in total. The current selected shape does not appear in the menu. The shapes mimic the rendering of the default filled in circle: the size is proportional to the number of occurrences, the kick appears as the shape, etc. 

Other changes: 
1. `#dom/shapes.js` now includes functions to calculate paths on the fly. 
2. The skewer text appears white for filled shapes or black for empty shapes. 

Test: 
1. [Filtered GDC track](http://localhost:3000/?genome=hg38&gene=ENST00000407796&mds3=GDC&filterObj={%22type%22:%22tvslst%22,%22in%22:true,%22join%22:%22%22,%22lst%22:[{%22type%22:%22tvs%22,%22tvs%22:{%22term%22:{%22id%22:%22case.diagnoses.age_at_diagnosis%22},%22ranges%22:[{%22start%22:10000,%22stop%22:20000,%22startinclusive%22:false,%22stopinclusive%22:true}],%22isnot%22:true}}]})
2. Checkout `filteredTkEx ` from `sjpp` and open http://localhost:3000/example.subtracks.html
3. [Scatter plot with shapes](http://localhost:3000/?noheader=1&mass={%22dslabel%22:%22PNET%22,%22genome%22:%22hg19%22,%22plots%22:[{%22chartType%22:%22sampleScatter%22,%22name%22:%22Methylome%20TSNE%22,%22colorTW%22:{%22id%22:%22TSNE%20Category%22},%20%22shapeTW%22:%20{%22term%22:%20{%22gene%22:%20%22KRAS%22,%20%22type%22:%20%22geneVariant%22}}}]}). Should remain uneffected. 

## Checklist

[Check each task](https://github.com/stjude/proteinpaint/wiki/Pull-Request-Checklist) that has been performed or verified to be not applicable.
- [x] Tests: added and/or passed unit and integration tests, or N/A
- [x] Todos: commented or documented, or N/A
- [x] Notable Changes: updated release.txt, prefixed a commit message with "fix:" or "feat:", added to an internal tracking document, or N/A

### CODE REVIEW

**Review by karishma-gangwani — DISMISSED** [review]
Looks great! Thanks @creilly8 

just adding a note here about the mode change from the y-axis to lollipops. 

@xzhou82 we should add the sub-tracks to the session files so I can create the link for mbmeta. I can't show the subtypes in subtracks right now.

File: client/mds3/leftlabel.variant.js [hunk 1]
```diff
@@ -166,6 +167,95 @@ function menu_variants(tk, block) {
...
+					.style('font-size', '20px')
+					.style('margin-top', '5px')
+
+				const shapesContainer = vectorGraphicsDiv
+					.append('div')
+					.style('display', 'flex')
+					.style('flex-wrap', 'wrap')
+					.style('width', 'max-content')
+
+				Object.entries(desiredShapes).forEach((val, idx) => {
```
Comment (xzhou82) [inline client/mds3/leftlabel.variant.js:230]: idx is not used. isn't it cleaner to use `for(const key in obj)`?

File: client/mds3/skewer.render.js [hunk 2]
```diff
@@ -125,28 +127,37 @@ export function skewer_make(tk, block) {
...
-		.attr('stroke', 'none')
-		.attr('d', d =>
-			d3arc()({
-				innerRadius: 0,
-				outerRadius: d.radius - 2,
-				startAngle: d.useNterm ? 0 : Math.PI,
-				endAngle: d.useNterm ? Math.PI : Math.PI * 2
-			})
-		)
+	const isEmptyCircle = tk.skewer?.shape?.[0] === 'emptyCircle' || false
```
Comment (xzhou82) [inline client/mds3/skewer.render.js:130]: no need for `|| false`

File: client/mds3/skewer.render.js [hunk 3]
```diff
@@ -125,28 +127,37 @@ export function skewer_make(tk, block) {
...
+
+	if (tk.skewer.shape && !tk.skewer.shape[0].includes('Circle')) {
```
Comment (xzhou82) [inline client/mds3/skewer.render.js:132]: if skewer.shape?.[0]?.includes(xx)

File: client/mds3/skewer.render.js [hunk 4]
```diff
@@ -125,28 +127,35 @@ export function skewer_make(tk, block) {
...
-		.attr('stroke', 'none')
-		.attr('d', d =>
-			d3arc()({
-				innerRadius: 0,
-				outerRadius: d.radius - 2,
-				startAngle: d.useNterm ? 0 : Math.PI,
-				endAngle: d.useNterm ? Math.PI : Math.PI * 2
-			})
-		)
+	const isEmptyShape = tk.skewer?.shape?.[1]?.isFilled == false
```
Comment (xzhou82) [inline client/mds3/skewer.render.js:130]: please comment on purpose of `[1]` and `[0]` from skewer.shape

File: client/mds3/skewer.render.js [hunk 5]
```diff
@@ -125,28 +127,35 @@ export function skewer_make(tk, block) {
...
+
+	if (tk.skewer.shape && !tk.skewer?.shape?.[0].includes('Circle')) {
+		renderSkewerShapes(tk, ss, discg)
+	} else {
```
Comment (xzhou82) [inline client/mds3/skewer.render.js:134]: add comment: i assume here it's using circle shape or not using shape?

File: client/mds3/skewer.render.js (hunk 4 above)
Comment (creilly8) [reply client/mds3/skewer.render.js:130]: The variant label script passes the object key (e.g. 'filledTriangle') as 0 and the object with the .path, .calculatedPath(), and .isFilled as 1. I use the key[0] to determine whether or not use a function or return certain arguments and the properties in the object[1] for rendering. 

I created isEmptyShape to keep the code a little cleaner. It's not necessary, just easier to read.

Comment (xzhou82) [reply client/mds3/skewer.render.js:130]: using .shape[0] is okay for quick prototyping, but `.shape={name, ...}` is more understandable and maintainable

Comment (creilly8) [reply client/mds3/skewer.render.js:130]: That's a valid point. I can make that change.

**Review by xzhou82 — APPROVED** [review]
thanks

**Pull request merged**